Test dismod

gma-uw opened the pull request:
There are 2 failing tests in this suite of tests, and they should fail because dismod doesn't seem to be working right.

gma-uw committed:
Initial checkin of test checking dismod execution results.

gma-uw committed:
removed some non-pytest code.

gma-uw committed:
Improved the error messages.

mbannick commented on tests/dismod_execution/dismod_tests.py:
```diff
@@ -0,0 +1,196 @@
+
+    return (truth, priors, node_effects, group_effects)
+
+def run_test(file_name, test_config, truth_in,
```
All your test names need to start with `test_` otherwise `pytest` won't see them. So this should be changed to `test_run` or something. And then the above test that doesn't actually have any asserts in it should not start with the name `test_`.

To skip running this test when you're on travis, for example, you would have:
```
def test_run(file_name, test_config, truth_in, start_from_truth, test_asymptotic, dismod):
    ...
```
It knows to skip this test in travis because of the `dismod` argument.

gma-uw replied:
create_database.py, dismod_tests.py, example_db.py are not tests. They are used by the other modules having names that start with test_   Is you comment still valid?

gma-uw committed:
Try a dismod arg

gma-uw committed:
Try a dismod arg

gma-uw committed:
Factored out a function.

gma-uw committed:
whatever

mbannick committed:
DISMODAT-705: Documentation (#311)

* restored docs from commit 20d1270952

* Refactored documentation

* Updated docs for cascade

* Updated docs for settings, context, jobmon, saver, cascade, and core.

* Documentation for other inputs.

* documented traditional cascade structure.

* documented traditional cascade structure.

* Docs for dismod api

* Fixed broken tests

* Finished with dismod io docs

* finished with docs

* Finished with docs.

mbannick committed:
Added doc packages

mbannick committed:
fixed merge conflicts

mbannick committed:
http domain

mbannick committed:
Added to requirements http domain

mbannick committed:
Update setup.py

mbannick committed:
Update requirements.txt

mbannick committed:
Update conf.py

mbannick committed:
Update setup.py

mbannick committed:
Update requirements.txt

mbannick committed:
Update index.rst

mbannick committed:
Update dismod-io.rst

mbannick committed:
Added overview picture

mbannick committed:
Merge branch 'develop' of https://github.com/ihmeuw/cascade-at into develop

mbannick committed:
Update requirements.txt

mbannick committed:
Fix for passing errors back in sample?

mbannick committed:
Update db.py

Brianepaulson committed:
Update install.rst

https://jira.ihme.washington.edu/browse/DISMODAT-766

Brianepaulson committed:
Update install.rst

Added a space between .[ihme, docs] and a note that these are optional arguments.

zhengp0 committed:
Merge pull request #312 from ihmeuw/feature/DISMODAT-768

DISMODAT-768: Fix for passing errors back in sample

gma-uw committed:
This fixes a fatal error in the drill without changing any results.

gma-uw committed:
Jobmon imports cause problems when running on a local machine without jobmon.
Should figure out how to get the jobmon switch to the top level call to enable/disable the imports.

gma-uw committed:
When missing_sexes set was empty, the print caused an error.

gma-uw committed:
Wrote a bit more information to the log files.

gma-uw committed:
On the local machine, a call to elmo logging caused a switch from the typical output streams to a socket, which results in log message errors.

gma-uw committed:
Avoid writing these log messages without content.

gma-uw committed:
These changes seem to be the minimal requirement to make a cascade with asymptotic statistics run correctly. It is not tested for a drill, or for the cv strategy for priors.

gma-uw committed:
When running on a local machine (not the cluster) the upload database connection fails.

gma-uw committed:
Added the arguments asymptotic and cv_priors to calling arguments, and changed the sampling logic to sample and predict given asymptotic statistics.
This logic has only been tested with the --asymptotic switch at the top level set.
It has not been tested when asymptotic statistics fail in dismod.

gma-uw committed:
Modified the tests to accomodate the asymptotic and cv_priors arguments and logic.

gma-uw committed:
Changes to make posteriors and priors for both asymptotic and simulate/sample work.

gma-uw committed:
Forgot to pass n_sim and n_pool into root fit.

gma-uw committed:
Better error message.

gma-uw committed:
Attempt to recover from a multiprocessing failure for simulate/sample.

gma-uw committed:
Improve log messages.

gma-uw committed:
Fixed bad switch in test.

gma-uw committed:
Merge branch 'test_dismod' of https://github.com/ihmeuw/cascade-at into test_dismod

zhengp0 commented on src/cascade_at/core/db.py:
```diff
@@ -62,15 +69,24 @@ def __getattr__(self, name):
+import sys
+
+if 'darwin' in sys.platform:    # gma Something, perhaps db_tools, is importing db_queries incorrectly causing a deprecation error
+    import sys
```
Already import `sys` outside the `if`

zhengp0 commented on src/cascade_at/core/db.py:
```diff
@@ -62,15 +69,24 @@ def __getattr__(self, name):
+
+if 'darwin' in sys.platform:    # gma Add logic to skip jobmon imports if jobmon switch is not set at top level call
+    LOG.warning("FIXME -- gma -- Add logic to skip jobmon imports if jobmon switch is not set at top level call")
+else:
```
where is the `db_queries` go? seem like when is not macOS, you will not have `db_queries` anymore.

gma-uw committed:
Small import change to make this run on Mac OS.

gma-uw committed:
Simple test of dismod_at fit, sample asymptotic, predict sample.

gma-uw committed:
Fixed import errors.

gma-uw committed:
Fixed misplaced indent on db_queries import.

gma-uw committed:
Test was calling dismod incorrectly on the cluster.

gma-uw committed:
This test compares the result of fit both, sample simulate fit both, and sample asymptotic.

gma-uw committed:
Improved comments.

gma-uw committed:
Fixed incorrect call to dismod_at.

gma-uw committed:
Fixed the sex covariate value with the test.

gma-uw committed:
Missplaced variable.

gma-uw committed:
Fixed dismod command error.

gma-uw committed:
dismod_at was called incorrectly on the cluster.

gma-uw committed:
Maybe this will work.

gma-uw committed:
Another misplaced dmdismod command.

gma-uw committed:
I have no idea why this is failing.

gma-uw committed:
I had forgotten the dismod argument to the test_1 function, which is why pytest was not skipping the test on the cluster.
